add blog post for the new wireguard mode

decathorpe opened the pull request:
I have never used hugo before, so I'm not sure that everything is 100% the way it needs to be (though it's reasonably similar to Jekyll, which I *have* used).

Feel free to modify / shorten / extend / adapt the text itself, if necessary. 🙃

Kriechi commented on src/content/posts/new-wireguard-mode/index.md:
```diff
@@ -0,0 +1,97 @@
+require taking any actions with elevated privileges.
+
+Connecting another device and routing its traffic through mitmproxy becomes as simple as connecting to a WireGuard VPN
+--- either by using the WireGuard configuration that is printed by mitmproxy, or by scanning the QR code that is shown
```
```suggestion
--- either by using the WireGuard configuration file generated by mitmproxy, or by scanning the QR code that is shown
```

decathorpe replied:
I'll change this with the next push, thanks.

Kriechi commented on src/content/posts/new-wireguard-mode/index.md:
```diff
@@ -0,0 +1,97 @@
+
+## How it works
+
+The core of this feature is implemented in the [`mitmproxy-wireguard`](https://pypi.org/project/mitmproxy-wireguard/)
```
will the main mitmproxy package pull in this dependency automatically (for all users), or do interested users need to install mitmproxy-wireguard explicitly?

decathorpe replied:
It's a new dependency and will be included by default. I added a sentence to clarify this.

Kriechi commented on src/content/posts/new-wireguard-mode/index.md:
```diff
@@ -0,0 +1,97 @@
+
+Connecting another device and routing its traffic through mitmproxy becomes as simple as connecting to a WireGuard VPN
+--- either by using the WireGuard configuration that is printed by mitmproxy, or by scanning the QR code that is shown
+in the mitmweb interface.
```
A screenshot would be great! Or even a 30sec gif/video from a phone scanning the QR code from mitmweb and it auto-configured the wireguard phone app.

mhils replied:
Video I recorded the other day:
https://user-images.githubusercontent.com/1019198/192087170-9bf30c93-a2cd-4c24-8075-78963f4bb567.mp4

Screenshot:
![image](https://user-images.githubusercontent.com/1019198/192087189-797b0844-5586-4510-a564-e28a27561042.png)

decathorpe replied:
Great, I'll try to embed these two files :)

Kriechi commented on src/content/posts/new-wireguard-mode/index.md:
```diff
@@ -0,0 +1,97 @@
+
+&nbsp;
+
+&nbsp;
```
Please remove these blank paragraphs.

decathorpe replied:
I copied these from an existing blog post, but sure, I'll remove them.

decathorpe commented:
I think I addressed all comments with the new version, please take another look.
I'm not sure if I embedded the video correctly, though, I don't know hugo syntax, I tried to base it off of the hugo docs.
